api/v1beta2/configuration_types.go

```diff
@@ -104,12 +106,32 @@ type Property struct {
 	Value string `json:"value,omitempty"`
 }
 
-// Backend stores the state in a Kubernetes secret with locking done using a Lease resource.
+// BackendType is an enum type for the Terraform backend types
+// +kubebuilder:validation:Enum=remote;artifactory;azurerm;consul;cos;etcd;etcdv3;gcs;http;kubernetes;manta;oso;pg;s3;swift
+type BackendType string
+
+// Backend describes the Terraform backend configuration
 type Backend struct {
 	// SecretSuffix used when creating secrets. Secrets will be named in the format: tfstate-{workspace}-{secretSuffix}
 	SecretSuffix string `json:"secretSuffix,omitempty"`
 	// InClusterConfig Used to authenticate to the cluster from inside a pod. Only `true` is allowed
 	InClusterConfig bool `json:"inClusterConfig,omitempty"`
+
+	// Inline allows users to use raw hcl code to specify their Terraform backend
+	Inline string `json:"inline,omitempty"`
+
+	// BackendType indicates which backend type to use. This field is needed for custom backend configuration.
+	BackendType BackendType `json:"backendType,omitempty"`
```

Comment:
Why do we make `BackendType` as a type, instead of a string?

Reply:
Actually, I want use `BackendType` as enum type. I think use string is ok as we can add `kubebuilder` tag to list all the options.